components/molecule/selectField/src/index.js

```diff
@@ -0,0 +1,80 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+
+import MoleculeField from '@s-ui/react-molecule-field'
+import MoleculeSelect from '@s-ui/react-molecule-select'
+
+const getErrorState = (success, error) => {
+  if (error) return true
+  if (success) return false
+}
+
+class MoleculeSelectField extends Component {
+  refSelect = React.createRef()
+  id = this.props.id || `MoleculeSelect-${+new Date()}`
+
+  handleClick = () => {
+    const {current: domSelect} = this.refSelect
+    if (domSelect) domSelect.focus()
+  }
+
+  render() {
+    const {id, refSelect, handleClick} = this
+    const {
+      label,
+      successText,
+      errorText,
+      helpText,
+      inline,
+      children, // eslint-disable-line
+      ...props
+    } = this.props
+    const errorState = getErrorState(successText, errorText)
+    return (
+      <MoleculeField
+        name={id}
+        label={label}
+        successText={successText}
+        errorText={errorText}
+        helpText={helpText}
+        inline={inline}
+        onClickLabel={handleClick}
+      >
+        <MoleculeSelect
+          refMoleculeSelect={refSelect}
+          errorState={errorState}
+          {...props}
+        >
+          {children}
+        </MoleculeSelect>
+      </MoleculeField>
+    )
+  }
+}
+
+MoleculeSelectField.displayName = 'MoleculeSelectField'
+
+MoleculeSelectField.propTypes = {
+  /** Text to be displayed as label */
+  label: PropTypes.string.isRequired,
+
+  /** used as label for attribute and Select element id */
+  id: PropTypes.string,
+
+  /** Success message to display when success state  */
+  successText: PropTypes.string,
+
+  /* onChange callback */
+  onChange: PropTypes.func,
+
+  /** Error message to display when error state  */
+  errorText: PropTypes.string,
+
+  /** Help Text to display */
+  helpText: PropTypes.string,
+
+  /** Boolean to decide if elements should be set inline */
+  inline: PropTypes.bool
+}
```

Comment:
Would be good to add this to the `demo`. Could we? I would like to see how this affects the UI of the component

Reply:
This `prop` is not really part of the final component specification. I decided to let it just in case there's some need of it in some projects but from a UX point of view it seems it's not a recommended way of using these components in a form.

Reply:
as this shouldn't have been implemented because design hasn't define this behaviour, we will let the prop as it is but without adding an example to our demos and creating a tech debt issue in order to remove it as soon as possible